### PR TITLE
Remove Command::Spellchecker and use DidYouMean for suggestions

### PR DESCRIPTION
**Pull request by p8**
Instead of generating our own "Maybe you meant ..." message, when a
generator or a server isn't found, use DidYouMean.
This commit adds a generic CorrectableError that could be used for
other commands.

By using DidYouMean we can delete the Command::Spellchecker.

## Before
image

## After
image

### CODE REVIEW

File: railties/lib/rails/commands/server/server_command.rb [hunk 1]
```diff
@@ -262,11 +262,10 @@ def rack_server_suggestion(server)
               Run `bin/rails server --help` for more options.
             MSG
           else
-            suggestion = Rails::Command::Spellchecker.suggest(server, from: RACK_SERVERS)
-            suggestion_msg = "Maybe you meant #{suggestion.inspect}?" if suggestion
-
+            error = CorrectableError.new("Could not find server '#{server}'", server, RACK_SERVERS)
```
Comment (kamipo) [inline railties/lib/rails/commands/server/server_command.rb:265]:
```suggestion
            error = CorrectableError.new("Could not find server '#{server}'.", server, RACK_SERVERS)
```

File: railties/lib/rails/generators.rb [hunk 2]
```diff
@@ -272,12 +272,12 @@ def invoke(namespace, args = ARGV, config = {})
           klass.start(args, config)
           run_after_generate_callback if config[:behavior] == :invoke
         else
-          options     = sorted_groups.flat_map(&:last)
-          suggestion  = Rails::Command::Spellchecker.suggest(namespace.to_s, from: options)
-          suggestion_msg = "Maybe you meant #{suggestion.inspect}?" if suggestion
+          options = sorted_groups.flat_map(&:last)
+          error   = Command::Base::CorrectableError.new("Could not find generator '#{namespace}'", namespace, options)
```
Comment (kamipo) [inline railties/lib/rails/generators.rb:276]:
```suggestion
          error   = Command::Base::CorrectableError.new("Could not find generator '#{namespace}'.", namespace, options)
```

Comment (p8) [reply railties/lib/rails/generators.rb]: I've add the dots.

**Comment by p8** [comment]
@kamipo I've also removed the extra blank line in the output, to keep it more with the original message.

**Pull request merged**